src/AspNetCoreModuleV2/AspNetCore/src/applicationinfo.cpp

```diff
@@ -218,26 +219,40 @@ APPLICATION_INFO::EnsureApplicationCreated(
         //
         if (!m_fAppOfflineFound)
         {
-
+            m_appStatus = APPLICATION_STATUS::FAIL;
             // Move the request handler check inside of the lock
             // such that only one request finds and loads it.
             // FindRequestHandlerAssembly obtains a global lock, but after releasing the lock,
             // there is a period where we could call
 
-            RETURN_IF_FAILED(FindRequestHandlerAssembly(struExeLocation));
+            FINISHED_IF_FAILED(FindRequestHandlerAssembly(struExeLocation));
 
             if (m_pfnAspNetCoreCreateApplication == NULL)
             {
-                RETURN_IF_FAILED(HRESULT_FROM_WIN32(ERROR_INVALID_FUNCTION));
+                FINISHED_IF_FAILED(HRESULT_FROM_WIN32(ERROR_INVALID_FUNCTION));
             }
 
-            RETURN_IF_FAILED(m_pfnAspNetCoreCreateApplication(m_pServer, pHttpContext->GetApplication(), &pApplication));
+            FINISHED_IF_FAILED(m_pfnAspNetCoreCreateApplication(m_pServer, pHttpContext->GetApplication(), &pApplication));
             pApplication->SetParameter(L"InProcessExeLocation", struExeLocation.QueryStr());
             m_pApplication = pApplication;
+            m_appStatus = APPLICATION_STATUS::RUNNING;
         }
     }
 
-    return S_OK;
+    Finished:
+    if (FAILED(hr))
+    {
+        // Log the failure and update application info to not try again
+        UTILITY::LogEventF(g_hEventLog,
```

Comment:
Doesn't seem like you are updating apllication info

Reply:
Isn't this log message repetitive?

Reply:
Just remove the comment. above the call to LogEventF.